lib/cpp_peripherial.dart

```diff
@@ -0,0 +1,306 @@
+import 'package:universal_chess_driver/central.dart';
+import 'package:universal_chess_driver/peripherial.dart';
+import 'package:universal_chess_driver/peripherial_client.dart';
+import 'package:universal_chess_driver/utils.dart';
+
+class CppPeripheralRound implements PeripherialRound {
+  String? get fen => null;
+  bool? get isSynchronized => null;
+}
+
+class CppPeripherial implements Peripherial {
+  final PeripherialClient _client;
+  final Central _central;
+  late PeripherialState _state;
+  List<String> _features = [];
+  List<String> _variants = [];
+
+  CppPeripherial(this._client, this._central) {
+    _client.recieve().listen(
+        (dataChunks) => onPeripheralCmd(String.fromCharCodes(dataChunks)));
+    transitionTo(new IterableExchangeState(
+        _central.features.iterator,
+        "feature",
+        IterableExchangeState(
+            _central.variants.iterator, "variant", IdleState(), this._variants),
+        this._features));
+  }
+
+  List<String> get features => _features;
+  List<String> get variants => _variants;
+  Central get central => _central;
+  PeripherialRound get round => new CppPeripheralRound();
+
+  void onPeripheralCmd(String cmd) {
+    print("proto: " + "Peripheral: " + cmd);
```

Comment:
`print()` are not welcome for production code. Add dome `onLog()` method to `Peripheral`?